internal/gke/deploy.go

```diff
@@ -956,11 +954,90 @@ func createServiceAccountsIAMPolicyRole(ctx context.Context, config CloudConfig)
 	})
 }
 
+// getConfigCluster computes the region and the cluster that manages all the
+// applications within a Service Weaver project, and the cluster .
+func getConfigCluster(ctx context.Context, config CloudConfig, cfg *config.GKEConfig) (string, string, error) {
+	region, name, err := getRunningConfigCluster(config)
+	if err != nil {
+		return "", "", err
+	}
+	if region == "none" {
+		// No config cluster. Set up a config cluster in the first region from the
+		// list of regions where to rollout the app, as specified by the user.
+		return cfg.Regions[0], applicationClusterName, nil
+	}
+
+	// Check if the config cluster actually exists.
+	exists, err := hasCluster(ctx, config, name, region)
+	if err != nil {
+		return "", "", err
+	}
+	if exists {
+		return region, name, nil
+	}
+	// The config cluster doesn't exist; disable ingress.
+	// TODO(rgrandl): we have to force the deletion of the ingress, because some
+	// other resources that depend on the ingress might be lingering around. We
+	// should delete all these resources as well.
+	_, err = runGcloud(config,
+		fmt.Sprintf("Disable multi-cluster ingress for cluster %q in %q",
+			name, region), cmdOptions{},
+		"container", "fleet", "ingress", "disable", "--force")
+
+	// The name of new config clusters will always be applicationClusterName.
+	name = applicationClusterName
+	if err != nil {
+		return "", "", err
+	}
+
+	// Set up the config cluster in the first region from the list of regions
+	// where to rollout the app, as specified by the user.
+	return cfg.Regions[0], name, nil
+}
+
+// getRunningConfigCluster returns information about the config cluster that
+// manages all the applications within a Service Weaver project.
+//
+// Returns none if no config cluster is running.
+func getRunningConfigCluster(config CloudConfig) (string, string, error) {
+	// Check if there is a config cluster already running.
+	out, err := runGcloud(config, "", cmdOptions{},
+		"container", "fleet", "ingress", "describe",
+		"--format=value(spec.multiclusteringress.configMembership)")
+	if err != nil {
+		// There is no config cluster.
+		return "none", "none", nil
+	}
+
+	// There is a config cluster entry. Extract the region where the config
```

Comment:
Can you simplify this to clearly separate legacy stuff from new. Also, we could check that the ingress cluster is one of ours, e.g.:

```
out = strings.TrimSuffix(out, "\n")
var clusterName, fName string
if strings.Contains(out, "serviceweaver-config") { // legacy config cluster name
  clusterName = "serviceweaver-config"
} else if strings.Contains(out, applicationClusterName) {
  clusterName = applicationClusterName
} else {
  // return error here
}
regionMatcher := fmt.Sprintf("projects/%s/locations/global/memberships/%s-%s", config.Project, clusterName)
if _, err := fmt.Sscanf(out, regionMatcher, &region); err != nil {
  ...
}
```

Reply:
Looks much better, thank you.